soroban-env-host/src/native_contract/token/storage_types.rs

```diff
@@ -4,6 +4,7 @@ use soroban_native_sdk_macros::contracttype;
 
 pub(crate) const INSTANCE_BUMP_AMOUNT: u32 = 34560; // 2 days
 pub(crate) const BALANCE_BUMP_AMOUNT: u32 = 518400; // 30 days
+pub(crate) const LIFETIME_THRESHOLD: u32 = 4320; // 6 hours
```

Comment:
I think we need to move this closer to each other. Currently we'd only bump balance if it's almost expired, which is not ideal. I think we should do something like 29 days/30 days or maybe even 29 days 18 hours/30 days. (i.e. bump once per 1 day/6 hours respectively)

Reply:
Yeah that's a good point. To make the potential bump more fair, I'm thinking the INSTANCE_BUMP_AMOUNT should have it's own threshold as well.

Reply:
Given that we can't go above max expiration, I'm not sure if it can be that different. Maybe we can have somewhat lower low watermark on the instance, as instance is accessed more frequently. So the high watermark can be reduced respectively.

Also it's a bit annoying that these values are non-upgradeable and have to depend on minimal config (otherwise these would fail). But I'm not sure if we have a good solution at this point...

Reply:
I don't think I understand your reasoning - Are you saying there's no reason for the instance to live longer than a balance without any interaction?

Reply:
Ah I see what you're saying. Do you think 7 days is enough for the instance? 

And re: non-upgradeable values - we could actually store the values in contract and give control to the admin, but that doesn't work for the XLM contract.

Reply:
> Do you think 7 days is enough for the instance?
That's basically a tradeoff - lower values mean lower max cost per user, higher values provide higher automatic contract live. Given that previously we had just 2 days bump for the instance, I think 7 days should be ok (and if token is not used for 7 days and admin doesn't care about it... then probably it actually isn't needed in the ledger).

Yeah, I don't think it's worth to make this configurable in the contract itself, as admins can always bump their instances as much as necessary (if needed).

Reply:
Updated